java/code/src/com/redhat/rhn/frontend/xmlrpc/system/provisioning/powermanagement/PowerManagementHandler.java

```diff
@@ -0,0 +1,403 @@
+     * @param loggedInUser the user
+     * @param serverId the requested server id
+     * @param action the action to execute
+     * @return 1 on success
+     *
+     * @xmlrpc.doc Execute power management actions
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("int", "serverId")
+     * @xmlrpc.param #param("string", "action")
+     * @xmlrpc.returntype #return_int_success()
+     */
+    // FIXME: and idea of a better name? executeAction(), ...
+    public int doPowerManagement(User loggedInUser, Integer serverId, String action) {
+        Server server = null;
+        try {
+            server = SystemManager.lookupByIdAndUser(serverId.longValue(), loggedInUser);
+        }
+        catch (LookupException e) {
+            throw new NoSuchSystemException();
+        }
+        ValidatorError error = null;
+        switch (action) {
+        case "PowerOn":
+            error = new CobblerPowerCommand(loggedInUser, server,
+                    CobblerPowerCommand.Operation.PowerOn).store();
+            break;
+        case "PowerOff":
+            error = new CobblerPowerCommand(loggedInUser, server,
+                    CobblerPowerCommand.Operation.PowerOn).store();
+            break;
+        case "Reboot":
+            error = new CobblerPowerCommand(loggedInUser, server,
+                    CobblerPowerCommand.Operation.PowerOn).store();
+            break;
+
+        default:
+            log.error("Invalid power management action: " + action);
+            throw new InvalidArgsException("Invalid power management action: " + action);
+        }
+        if (error != null) {
+            log.error("Power management action " + action + "failed");
+            throw new PowerManagementOperationFailedException(error.getMessage());
+        }
+        log.info("Power management action " + action + "succeeded");
+        return 1;
+    }
+
+    /**
+     * Execute power management actions
+     *
+     * @param loggedInUser the user
+     * @param nameIn the cobbler name prefix
+     * @param action the action to execute
+     * @return 1 on success
+     *
+     * @xmlrpc.doc Execute power management actions
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param("string", "name")
+     * @xmlrpc.param #param("string", "action")
+     * @xmlrpc.returntype #return_int_success()
+     */
+    // FIXME: and idea of a better name? executeAction(), ...
+    public int doPowerManagement(User loggedInUser, String nameIn, String action) {
+        ensureOrgAdmin(loggedInUser);
+        ValidatorError error = null;
+        switch (action) {
+        case "PowerOn":
+            error = new CobblerPowerCommand(loggedInUser, nameIn,
+                    CobblerPowerCommand.Operation.PowerOn).store();
+            break;
+        case "PowerOff":
+            error = new CobblerPowerCommand(loggedInUser, nameIn,
+                    CobblerPowerCommand.Operation.PowerOn).store();
```

Comment:
its using `PowerOn` in all 3 cases right now. Same as above.

Reply:
Ups :-)